crates/ruff_linter/src/rules/refurb/rules/unnecessary_from_float.rs

```diff
@@ -288,7 +288,22 @@ fn handle_non_finite_float_special_case(
     };
     as_non_finite_float_string_literal(float_arg)?;
 
-    let replacement_arg = checker.locator().slice(float_arg).to_string();
+    let mut replacement_arg = checker.locator().slice(float_arg).to_string();
+    let trimmed = replacement_arg.trim().to_ascii_lowercase();
+    if trimmed.contains("nan") && trimmed.contains('-') {
+        if let Some(nan_idx) = trimmed.find("nan") {
+            let mut chars: Vec<char> = replacement_arg.chars().collect();
+            let mut i = 0;
+            while i < nan_idx {
+                if chars[i] == '-' {
+                    chars.remove(i);
+                    continue;
+                }
+                i += 1;
+            }
+            replacement_arg = chars.into_iter().collect();
+        }
+    }
```

Comment:
Does this work for `float("\x2dnan")` or `float("\N{HYPHEN-MINUS}nan")`? Cf. #16771.

Reply:
Yes, it covers both of those cases